pkg/hyper/client.go

```diff
@@ -667,3 +665,90 @@ func (c *Client) AttachContainer(containerID string, stdin io.Reader, stdout, st
 	}
 	return nil
 }
+
+// ExecInSandbox exec a command in sandbox with specified io, tty and timeout
+func (c *Client) ExecInSandbox(sandboxID string, cmd []string, stdin io.Reader, stdout, stderr io.WriteCloser, tty bool, timeout time.Duration) error {
+	var (
+		ctx    context.Context
+		cancel context.CancelFunc
+	)
+
+	if timeout > 0 {
+		ctx, cancel = getContextWithTimeout(timeout)
+	} else if timeout == 0 {
+		ctx, cancel = getContextWithCancel()
+	} else {
+		return fmt.Errorf("timeout should be non-negative")
+	}
+	defer cancel()
+
+	stream, err := c.client.ExecVM(ctx)
+	if err != nil {
+		return err
+	}
+	if err = stream.Send(&types.ExecVMRequest{
+		PodID:   sandboxID,
+		Command: cmd,
+	}); err != nil {
+		return err
+	}
+
+	var recvStdoutError chan error
+	if stdout != nil || stderr != nil {
+		recvStdoutError = promiseGo(func() (err error) {
+			for {
+				out, err := stream.Recv()
+				if out != nil {
+					if len(out.Stdout) > 0 && stdout != nil {
+						nw, ew := stdout.Write(out.Stdout)
+						if ew != nil {
+							return ew
+						}
+						if nw != len(out.Stdout) {
+							return io.ErrShortWrite
+						}
+					}
+				}
+				if err == io.EOF {
+					break
+				}
+				if err != nil {
+					return err
+				}
+			}
+			return nil
+		})
+	}
+
+	if stdin != nil {
+		go func() error {
+			defer stream.CloseSend()
+			buf := make([]byte, 32)
```

Comment:
Is 32 bytes buffer big enough? I'm not sure if there's other case we want to use `ExecInSandbox` with more input data .